### PR TITLE
Docs: Update contribution guides

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:

Changes to tooling or processes haven't been reflected in the contributing guidelines as there are still mentions of yarn and no explanation of how to run nx commands or basic info on how to contribute to the docs website.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,12 +32,16 @@ This is a mono-repo of NPM packages to help plugin developers extend Grafana in
 | Create Plugin | Scaffold a Grafana Plugin | [Link](./packages/create-plugin/README.md) | [![npm](https://img.shields.io/npm/v/@grafana/create-plugin)](https://www.npmjs.com/package/@grafana/create-plugin) | [![npm](https://img.shields.io/npm/dw/@grafana/create-plugin)](https://npmcharts.com/compare/@grafana/create-plugin?interval=7) |
 | Sign Plugin   | Sign a Grafana Plugin     | [Link](./packages/sign-plugin/README.md)   | [![npm](https://img.shields.io/npm/v/@grafana/sign-plugin)](https://www.npmjs.com/package/@grafana/sign-plugin)     | [![npm](https://img.shields.io/npm/dw/@grafana/sign-plugin)](https://npmcharts.com/compare/@grafana/sign-plugin?interval=7)     |
 
+### Overview
+
+This Mono-repo uses [NPM](https://nodejs.org) for package management, [NX](https://nx.dev/) to efficiently orchestrate tasks across the codebase, and [Auto](https://intuit.github.io/auto/) for streamlined and automated package publishing. We've carefully chosen and integrated these technologies to enhance development workflows. Before diving into the codebase, make sure to consult the [contributing guide](./CONTRIBUTING.md) for a smooth collaboration experience.
```
Comment (leventebalogh) [inline README.md:37]: 👏

File: docusaurus/website/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    │   ├── components
+    │   ├── css
+    │   ├── pages
+    │   └── theme
+    ├── static/ // folder for static assets
+    │   ├── font
+    │   └── img
+    ├── docusaurus.config.js
+    └── package.json
+```
```
Comment (leventebalogh) [inline docusaurus/website/CONTRIBUTING.md:51]: This is super useful. 🙏

**Review by leventebalogh — APPROVED** [review]
Amazing! 🥇

File: packages/sign-plugin/CONTRIBUTING.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ There are a collection of [commands](#commmands) to assist with developing `sign
 
 ### Commmands
 
-Below are the main commands used for developing `sign-plugin`. They can be run by either `yarn workspace @grafana/sign-plugin run <name_of_command>` or navigating to `packages/sign-plugin` and running the command directly as detailed below.
+Below are the main commands used for developing `sign-plugin`. Below are the main commands used for developing `sign-plugin`. They can be run by either `npx nx run @grafana/sign-plugin:<name_of_command>`, `npm run <name_of_command> -w @grafana/sign-plugin` or navigating to `packages/sign-plugin` and running the command directly as detailed below.
```
Comment (sympatheticmoose) [inline packages/sign-plugin/CONTRIBUTING.md:36]: duplicate "Below are the main commands used for developing `sign-plugin`."

File: docusaurus/website/CONTRIBUTING.md
Comment (sympatheticmoose) [inline docusaurus/website/CONTRIBUTING.md:1]: wdyt about adding the actions to push to dev developer portal?

File: README.md [hunk 4]
```diff
@@ -30,12 +32,16 @@ This is a mono-repo of NPM packages to help plugin developers extend Grafana in
...
+
 ### Additional resources
 
-📖 Learn from tutorials and documenation in the [Grafana developer portal](https://grafana.com/developers).  
-✨ Gain inspiration from our [plugin examples](https://github.com/grafana/grafana-plugin-examples/) to get started quickly and implement new features in your plugin.  
-🛠️ Use the [Grafana plugin SDK for Go](https://github.com/grafana/grafana-plugin-sdk-go) to simplify the development of backend components.  
-✅ Ensure your plugin is ready for publishing to the [Grafana plugin catalog](https://grafana.com/grafana/plugins/) with our [validator](https://github.com/grafana/plugin-validator/) tool.  
+📖 Learn from tutorials and documentation in the [Grafana developer portal](https://grafana.com/developers).
```
Comment (sympatheticmoose) [inline README.md:41]: removing whitespace means this is one paragraph, intentional?
Screenshot 2023-11-09 at 17 19 32

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -135,6 +136,7 @@ The main difference between code contributions and documentation contributions i
 To contribute code:
 
 - [Set up the project](#project-setup).
+- Review the CONTRIBUTING.md file for the package/s you are working on.
```
Comment (sympatheticmoose) [inline CONTRIBUTING.md:139]: In this section we could probably highlight that we request issues before PRs

File: CONTRIBUTING.md
Comment (sympatheticmoose) [inline CONTRIBUTING.md:1]: In `Report an Error or Bug` we talk about labels which don't exist `needs-X` and `critical`

**Review by sympatheticmoose — COMMENTED** [review]
Great work @jackw  👏 this clears up a lot of things. 

Helpfully late to the party with some additional feedback 😅

**Pull request merged**